### PR TITLE
Feature/1057 implement tables

### PR DESCRIPTION
**Pull request by loic294**
# Description

**Make sure the PR is against the `develop` branch (see [Contributing](https://github.com/nccgroup/ScoutSuite/blob/master/CONTRIBUTING.md)).**

Please include a summary of the change(s) and which issue(s) it addresses. Please also include relevant motivation and context.

Fixes # (issue)

## Type of change

Select the relevant option(s):

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works (optional)
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by SamuelMeilleur — CHANGES_REQUESTED** [review]
image
About the style:
- The background of the table should be the same as the app rather than white.
- The icons in the severity columns aren't centered (can be done with `display: flex;` and `align-items: center`.

**Review by SamuelMeilleur — CHANGES_REQUESTED** [review]

File: ScoutSuite/frontend/src/components/SeverityLabel/index.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import './style.scss';
+
+const propTypes = {
+  severity: PropTypes.string.isRequired,
+};
+
+const SeverityLabel = props => {
+  const { severity } = props;
+
+  if (severity === 'critical') {
```
[the remark this answers is not in the document]
Comment (loic294) [reply ScoutSuite/frontend/src/components/SeverityLabel/index.js:16]: Yes, totally! Copy/Paste is just so easy sometimes 🥺 I will change it for an object that contains the text/icon.

File: ScoutSuite/frontend/src/components/Table/index.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  const cols = columns.map((item) => {
+    let col = {
+      Header: item.name,
+      accessor: item.key,
+    };
+    if (formatters[item.key]) col.Cell = formatters[item.key];
+    if (sortBy[item.key])
+      col.sortType =
+        typeof sortBy[item.key] === 'function'
```
[the remark this answers is not in the document]
Comment (loic294) [reply ScoutSuite/frontend/src/components/Table/index.js:33]: Oh, good catch! I am supposed to "memoize" the content value if it's a function but had a problem with that so I removed it and forgot to add it back. I will fix that.

**Comment by loic294** [comment]
> image
> 
> About the style:
> 
> - The background of the table should be the same as the app rather than white.
> 
> - The icons in the severity columns aren't centered (can be done with `display: flex;` and `align-items: center`.

- Not sure to understand the why the background colour would be something else than white? That's what I had in the design. 

- `display: flex;` actually messes up the with of the columns but I will use which is specific for tables 

```
td 
{
    text-align: center; 
    vertical-align: middle;
} 
```

**Review by SamuelMeilleur — APPROVED** [review]

**Pull request merged**